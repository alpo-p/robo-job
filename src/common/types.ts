import { Ionicons } from '@expo/vector-icons'

export type IoniconsIconNames = keyof typeof Ionicons.glyphMap

export interface IJobPostCard {
  id: string
  companyName: string
  jobTitle: string
  logoUrl: string
  pictureUrl: string
  location: string
  tags: string[]
  shortDescription: string
  longDescription: string
  deadline: string
  aboutTheCompany: string
  linkToWebsite: string
}
