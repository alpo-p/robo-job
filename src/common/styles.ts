export default {
  backgroundColor: '#000000',
  white: '#ffffff',
  darkBackgroundColor: '#222222',
  darkGrey: '#404040',
  mediumDarkGrey: '#d0d0d0',
  lightGrey: '#e0e0e0',

  successGreen: '#087c61',

  primaryColorBase: '#186db6',
  primaryColorDarkened10: '#00549D',
  primaryColorLightened10: '#3287D0',
  primaryColorDarkened20: '#003A83',
  primaryColorLightened20: '#4BA0E9',
  primaryColorDarkened30: '#00216A',

  backgroundWhite: 'rgb(242, 242, 242)',
  darkText: 'rgb(28, 28, 30)',
  whiteText: 'rgba(255, 255, 255, 0.87)',
  commonSize: 24,
  biggerCommonSize: 38,
}
