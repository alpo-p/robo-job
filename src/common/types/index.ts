export interface IJobPostCard {
  id: string
  companyName: string
  jobTitle: string
  logoUrl: string
  pictureUrl: string
  location: string
  tags: string[]
  shortDescription: string
  longDescription: string
  deadline: string
  aboutTheCompany: string
  linkToWebsite: string
}

export type MessageTypes = 'robo' | 'recruiter' | 'user'

export interface Message {
  typeOfMessage: MessageTypes
  text: string
}

export interface IndividualChat {
  jobPostId: string
  messages: Message[]
}

export type Chats = IndividualChat[]
