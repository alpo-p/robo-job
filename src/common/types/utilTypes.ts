import { Ionicons } from '@expo/vector-icons'

export type IoniconsIconNames = keyof typeof Ionicons.glyphMap

export type SetStringStateType = React.Dispatch<React.SetStateAction<string[]>>
