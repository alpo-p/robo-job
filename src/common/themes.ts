import {
  DarkTheme as DefaultDarkTheme,
  DefaultTheme,
} from '@react-navigation/native'
import styles from './styles'

export type ThemeColors = {
  primary: string
  background: string
  card: string
  text: string
  border: string
  notification: string
}

export const LightTheme = {
  ...DefaultTheme,
  colors: {
    ...DefaultTheme.colors,
    primary: styles.primaryColor,
    text: styles.darkText,
    icon: '#121212',
    background: styles.white,
  },
}

export const DarkTheme = {
  ...DefaultDarkTheme,
  colors: {
    ...DefaultDarkTheme.colors,
    primary: styles.primaryColor,
    text: styles.whiteText,
    card: '#121212',
    border: '#121212',
    icon: styles.whiteText,
  },
}
