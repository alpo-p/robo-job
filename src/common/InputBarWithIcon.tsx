import { Ionicons } from '@expo/vector-icons'
import { useTheme } from '@react-navigation/native'
import React from 'react'
import { TextInput, useColorScheme, View } from 'react-native'
import styles from './styles'
import { IoniconsIconNames } from './types/utilTypes'

interface P {
  value: string
  // eslint-disable-next-line no-unused-vars
  setValue: (v: string) => void
  iconName: IoniconsIconNames
  placeholderText?: string
}

// TODO: refactor
export default ({
  value,
  setValue,
  iconName,
  placeholderText = 'Search',
}: P) => {
  const { colors } = useTheme()
  const scheme = useColorScheme()
  // TODO: refactor bgColor and placeholder color to themes
  const bgColor =
    scheme === 'dark' ? styles.lightBgDarkMode : styles.lightBgLightMode
  const placeholderColor =
    scheme === 'dark' ? styles.lighterWhiteText : styles.lighterDarkText
  return (
    <View
      style={{
        flexDirection: 'row',
        backgroundColor: bgColor,
        margin: 16,
        borderRadius: 8,
        height: 8 * 4,
      }}
    >
      <Ionicons
        style={{
          padding: 8,
        }}
        name={iconName}
        color={colors.primary}
        size={15}
      />
      <TextInput
        style={{
          color: colors.text,
          width: '70%',
        }}
        value={value}
        onChangeText={setValue}
        placeholder={placeholderText}
        placeholderTextColor={placeholderColor}
      />
    </View>
  )
}
