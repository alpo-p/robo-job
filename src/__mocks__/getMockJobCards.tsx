import { IJobPostCard } from '../common/types'
import getMockJobPostCard from './getSingleMockJobCard'

export default function getMockJobCards(n = 10): IJobPostCard[] {
  const mockJobPostCards = [...Array(n).keys()].map(i => {
    const mockCard = getMockJobPostCard()
    mockCard.id = `${i}`
    mockCard.jobTitle = `Software developer #${i}`
    return mockCard
  })
  return mockJobPostCards
}
