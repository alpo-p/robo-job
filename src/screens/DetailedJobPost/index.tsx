import { RouteProp, useNavigation } from '@react-navigation/native'
import React from 'react'
import { ScrollView, View } from 'react-native'
import { SafeAreaView } from 'react-native-safe-area-context'
import { IJobPostCard } from '../../common/types'
import BasicInfo from './components/BasicInfo'
import BottomButtons from './components/BottomButtons'
import AboutThePosition from './components/AbouThePosition'
import PictureAndLogo from './components/PictureAndLogo'
import AboutTheCompany from './components/AboutTheCompany'
import GoBackButtonHeavy from '../../common/components/GoBackButtonHeavy'
import { VerticalSpaceOf24 } from '../../common/components/VerticalSpaceOf24'

interface P {
  route: RouteProp<{ params: { jobPost: IJobPostCard } }, 'params'>
}

const DetailedJobCard: React.FC<P> = ({
  route: {
    params: { jobPost },
  },
}) => {
  const navigation = useNavigation()
  const navigateBackToSearch = () => navigation.goBack()

  return (
    <SafeAreaView>
      <GoBackButtonHeavy onPress={navigateBackToSearch} />
      <ScrollView>
        <PictureAndLogo
          pictureUrl={jobPost.pictureUrl}
          logoUrl={jobPost.logoUrl}
        />
        <View
          style={{
            marginTop: 64,
            marginBottom: 50,
            paddingLeft: 16,
            paddingRight: 24,
          }}
        >
          <BasicInfo jobPost={jobPost} />
          <VerticalSpaceOf24 />
          <AboutThePosition description={jobPost.longDescription} />
          <VerticalSpaceOf24 />
          <AboutTheCompany jobPost={jobPost} />
        </View>
      </ScrollView>
      <BottomButtons />
    </SafeAreaView>
  )
}

export default DetailedJobCard
