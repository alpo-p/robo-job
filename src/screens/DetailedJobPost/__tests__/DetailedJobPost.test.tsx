import { RouteProp } from '@react-navigation/native'
import React from 'react'
import { IJobPostCard } from '../../../common/types'
import create from '../../../testUtils/rendererCreateWithNavigation'
import getMockJobCards from '../../../__mocks__/getMockJobCards'
import DetailedJobPost from '../index'

describe('DetailedJobPost', () => {
  const prop = {
    params: { jobPost: getMockJobCards()[0] },
  } as unknown as RouteProp<{ params: { jobPost: IJobPostCard } }, 'params'>

  it('should match snapshot', () => {
    const tree = create(<DetailedJobPost route={prop} />)
    expect(tree).toMatchSnapshot()
  })
})
