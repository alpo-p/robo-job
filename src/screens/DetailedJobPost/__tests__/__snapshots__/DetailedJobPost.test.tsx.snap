// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailedJobPost should match snapshot 1`] = `
<RNCSafeAreaView>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "flex-start",
        "marginLeft": 8,
        "opacity": 1,
        "position": "absolute",
        "top": 48,
        "zIndex": 1,
      }
    }
  >
    <Text />
  </View>
  <RCTScrollView>
    <View>
      <View
        accessibilityIgnoresInvertColors={true}
        style={
          Object {
            "height": 280,
          }
        }
      >
        <Image
          resizeMode="stretch"
          source={
            Object {
              "uri": "https://thumbs.dreamstime.com/b/close-up-business-people-congratulating-one-another-team-building-exercise-54970426.jpg",
            }
          }
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "height": 280,
                "width": undefined,
              },
              undefined,
            ]
          }
        />
        <Image
          source={
            Object {
              "uri": "https://ih1.redbubble.net/image.659413411.3655/st,small,507x507-pad,600x600,f8f8f8.u1.jpg",
            }
          }
          style={
            Object {
              "alignSelf": "center",
              "borderColor": "white",
              "borderRadius": 56,
              "borderWidth": 4,
              "bottom": -56,
              "height": 112,
              "position": "absolute",
              "width": 112,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "marginBottom": 50,
            "marginTop": 64,
            "paddingLeft": 16,
            "paddingRight": 24,
          }
        }
      >
        <View>
          <View
            style={
              Object {
                "alignItems": undefined,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000000",
                  "fontFamily": "DamascusBold",
                  "fontSize": 20,
                  "textTransform": "none",
                }
              }
            >
              E Corp
            </Text>
            <View
              style={
                Object {
                  "marginTop": 4,
                }
              }
            />
            <Text
              style={
                Object {
                  "color": "#000000",
                  "fontFamily": "Damascus",
                  "fontSize": 20,
                  "textTransform": "none",
                }
              }
            >
              Digital Marketing Manager
            </Text>
          </View>
          <View
            style={
              Object {
                "marginTop": 16,
              }
            }
          />
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <Text />
            <Text
              style={
                Object {
                  "color": "#000000",
                  "fontFamily": "DamascusLight",
                  "fontSize": 14,
                  "textTransform": "uppercase",
                }
              }
            >
              Helsinki
            </Text>
          </View>
          <View
            style={
              Object {
                "marginTop": 16,
              }
            }
          />
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <Text />
            <Text
              style={
                Object {
                  "color": "#000000",
                  "fontFamily": "DamascusLight",
                  "fontSize": 14,
                  "textTransform": "none",
                }
              }
            >
              Deadline in 
              7
               days
            </Text>
          </View>
          <View
            style={
              Object {
                "marginTop": 24,
              }
            }
          />
          <View
            style={
              Object {
                "width": "100%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000000",
                  "fontFamily": "DamascusLight",
                  "fontSize": 14,
                  "lineHeight": 24,
                  "textAlign": "justify",
                  "textTransform": "none",
                }
              }
            >
              As a Digital Marketing Manager you will be responsible for digital marketing content creation, company newsletter, web pages, social media, photographing and graphic design for web and print media. You will be accompanies by a happy group of hard-working and nice people with good values and great benefits
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "marginTop": 24,
            }
          }
        />
        <View>
          <Text
            style={
              Object {
                "color": "#000000",
                "fontFamily": "Damascus",
                "fontSize": 20,
                "textTransform": "none",
              }
            }
          >
            About the position
          </Text>
          <View
            style={
              Object {
                "marginBottom": 8,
              }
            }
          />
          <Text
            style={
              Object {
                "color": "#000000",
                "fontFamily": "DamascusLight",
                "fontSize": 14,
                "lineHeight": 24,
                "textAlign": "justify",
                "textTransform": "none",
              }
            }
          >
            We are searching for a passionate and ambitious MARKETING MANAGER to take on the exciting challenge of managing the company’s marketing activities in their home market - Finland!
  
As the successful candidate, we trust that you will bring relevant marketing experience with proven success, along with a strong can-do attitude and a banquet of fresh ideas! In this marketing role, you will have the opportunity to influence the direction of some of the most respected brands! 
  
You will get to collaborate with colleagues around the world, create campaigns, participate at events, impact product design projects, communicate with like-minded marketing professionals and build brand communities – amongst many other exciting tasks!
          </Text>
        </View>
        <View
          style={
            Object {
              "marginTop": 24,
            }
          }
        />
        <View>
          <Text
            style={
              Object {
                "color": "#000000",
                "fontFamily": "Damascus",
                "fontSize": 20,
                "textTransform": "none",
              }
            }
          >
            About 
            E Corp
             
          </Text>
          <View
            style={
              Object {
                "marginBottom": 8,
              }
            }
          />
          <Text
            style={
              Object {
                "color": "#000000",
                "fontFamily": "DamascusLight",
                "fontSize": 14,
                "lineHeight": 24,
                "textAlign": "justify",
                "textTransform": "none",
              }
            }
          >
            E Corp is a fast and responsible loan provider with over 500 000 customers in Finland, Sweden and Lithuania. The company was recently granted a specialized banking license by The European Central Bank (ECB) and is currently establishing a new business to enable international growth and a continuous launch of new products.

E Corp’s unique competitive strength is based on a fully automated lending process with a proprietary scoring system that enables fast, secure and personalized loan offers in a responsible way
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "backgroundColor": "#000000",
        "bottom": 40,
        "flexDirection": "row",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text />
    </View>
    <View
      style={
        Object {
          "marginLeft": 24,
        }
      }
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text />
    </View>
    <View
      style={
        Object {
          "marginLeft": 24,
        }
      }
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text />
    </View>
  </View>
</RNCSafeAreaView>
`;
